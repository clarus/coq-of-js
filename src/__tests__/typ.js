// @flow
import {compileAndPrint} from "../compiler/index.js";

it("shows an error message on unhandled types", () => {
  expect(compileAndPrint(`const a: number | string = 12;`)).toMatchSnapshot();
});
