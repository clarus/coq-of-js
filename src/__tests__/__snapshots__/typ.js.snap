// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows an error message on unhandled types 1`] = `
"> 1 | const a: number | string = 12;
    |         ^^^^^^^^^^^^^^^

Unhandled syntax:
{
  \\"type\\": \\"UnionTypeAnnotation\\",
  \\"start\\": 9,
  \\"end\\": 24,
  \\"loc\\": {
    \\"start\\": {
      \\"line\\": 1,
      \\"column\\": 9
    },
    \\"end\\": {
      \\"line\\": 1,
      \\"column\\": 24
    }
  },
  \\"types\\": [
    {
      \\"type\\": \\"NumberTypeAnnotation\\",
      \\"start\\": 9,
      \\"end\\": 15,
      \\"loc\\": {
        \\"start\\": {
          \\"line\\": 1,
          \\"column\\": 9
        },
        \\"end\\": {
          \\"line\\": 1,
          \\"column\\": 15
        }
      }
    },
    {
      \\"type\\": \\"StringTypeAnnotation\\",
      \\"start\\": 18,
      \\"end\\": 24,
      \\"loc\\": {
        \\"start\\": {
          \\"line\\": 1,
          \\"column\\": 18
        },
        \\"end\\": {
          \\"line\\": 1,
          \\"column\\": 24
        }
      }
    }
  ]
}"
`;
